### PR TITLE
Sample Logs should store units, besides value

### PR DESCRIPTION
**Pull request by jmborr**
Work done:
- [x] added units attribute to sample logs
- [x] `Hydraworkspace` can transfer units to the sample logs
- [x] `NeXusConvertingApp` can transfer log units to SampleLogs
- [x] `HydraWorkspace` can save log units to the project file
- [x] `HydraWorkspace` can read log units from the project file
- [x] `SampleLogs.get_pointlist` creates point lists in units of mili-meters

Refs #629

### CODE REVIEW

**Review by peterfpeterson — CHANGES_REQUESTED** [review]
A couple of minor comments

File: pyrs/core/nexus_conversion.py [hunk 1]
```diff
@@ -72,6 +72,13 @@ def calculate_sub_run_time_average(log_property, time_filter) -> float:
 
 
 class Splitter:
+    r"""
```
[the remark this answers is not in the document]
Comment (jmborr) [reply pyrs/core/nexus_conversion.py:75]: Yes, [this link explains](https://stackoverflow.com/questions/37019744/is-there-a-consensus-what-should-be-documented-in-the-classes-and-init-docst#:~:text=If%20the%20intention%20of%20the,public%20aspects%20of%20the%20object.) under what conditions should the docstring of ` __init__` be included in the class' docstring.
After reading it, it seems that the docstring should go under `__init__` becuase `Splitter` is not part of the public interface

[the remark this answers is not in the document]
Comment (jmborr) [reply pyrs/core/nexus_conversion.py:75]: I'm changing the location

File: pyrs/dataobjects/sample_logs.py [hunk 2]
```diff
@@ -497,9 +525,12 @@ def get_pointlist(self, subruns=None) -> 'PointList':
         if missing:
             raise ValueError('Failed to find positions in logs. Missing {}'.format(', '.join(missing)))
 
+        # Check the units are in mili meters
+        factor = 1000.0 if self._units.get(VX, 'mm') == 'm' else 1.0
```
[the remark this answers is not in the document]
Comment (jmborr) [reply pyrs/dataobjects/sample_logs.py:529]: Yep, that's why I only check for `VX`

**Review by peterfpeterson — APPROVED** [review]

**Comment by peterfpeterson** [comment]
Fixes #629

**Pull request merged**